react/features/base/redux/functions.js

```diff
@@ -111,6 +111,29 @@ function _set(
 
 /* eslint-enable max-params */
 
+/**
+ * Returns redux feature's state from the specified {@code stateful} which is
+ * presumed to be related to the redux state (e.g. the redux store, the redux
+ * {@code getState} function). The third option for {@code stateful} is the
+ * redux {@code feature}'s state in which case the whole processing will be
+ * a NOP. It is assumed that if the state returned by {@link getState} for given
+ * {@code stateful} has no value for {@code feature} key it is probably
+ * the feature's state.
+ *
+ * @param {Function|Object} stateful - The entity such as the redux store or the
+ * redux {@code getState} function from which the redux state is to be
+ * returned or the redux feature's state object directly.
+ * @param {string} feature - The Redux feature's name for which the state will
+ * be obtained.
+ * @returns {Object} The redux feature's state for given name obtained from
+ * given {@code stateful}.
+ */
+export function toFeatureState(stateful: Function | Object, feature: string) {
+    const state = toState(stateful);
+
+    return state[feature] ? state[feature] : state;
```

Comment:
since my previous comment was attached to the old PR, I'm making it here again, because I don't remember the discussion, sorry -- when would returning the entire state be helpful? I fail to see this usecase.

Reply:
I've tried hard to explain that in the docs, but I'll try to improve. The function will accept 3 options as the `stateful` argument:

1. The Redux store's `getState` function
2. The whole Redux state returned by `getState`
3. Specific feature's state retrieved from the Redux store state through: `state[feature]`

If the key doesn't exit it means it's the option `#3` (there's no guarantee it's not some garbage though).

Reply:
Yes, that I see, but what I don't know is why and when `#3` would make any sense. If I request the state for a feature but I make a typo this function gives me something I didn't ask for. That IMHO is not right. If it doesn't exist it should return `undefined`.

Reply:
The purpose of that method was to be able to call it also from a reducer which gets the exact feature's state:
https://github.com/jitsi/jitsi-meet/pull/3153/commits/1d55555cbcffa08ed25343ab630ca05b1763ff90#diff-9081be03514c8917e9eb1a4551893273R114

Maybe I need to think of something else but returning `undefined` will not work for me

Reply:
I see. IMHO it's very counter intuitive and I'd just use `toState` and index accordingly, as we have already been doing.

Reply:
I'm not sure if I explained you correctly what I need, but I went with different approach by just extracting a local function in the reducer to avoid duplication and that's it.

I don't have any better idea to be able to deal with a `stateful` which can also be a reducer's state or a Redux `state`/`getState` function. What you write about `toState` has no use for me at all, because I have the state in the reducer already. I just wanted to have a method which would accept any of the 3.